STUDIO/org.openl.security.acl/src/org/openl/security/acl/utils/AclPathUtils.java

```diff
@@ -0,0 +1,110 @@
+package org.openl.security.acl.utils;
+
+import java.util.List;
+
+import org.openl.rules.project.abstraction.AProjectArtefact;
+import org.openl.rules.project.impl.local.LocalRepository;
+import org.openl.rules.project.impl.local.ProjectState;
+import org.openl.rules.repository.api.FileData;
+import org.openl.rules.workspace.dtr.impl.FileMappingData;
+import org.openl.util.StringUtils;
+
+public final class AclPathUtils {
+
+    private AclPathUtils() {
+    }
+
+    /**
+     * Concatenates two paths.
+     *
+     * @param path1 first path
+     * @param path2 second path
+     * @return concatenated path
+     */
+    public static String concatPaths(String path1, String path2) {
+        if (StringUtils.isBlank(path1)) {
+            return path2;
+        }
+        if (StringUtils.isBlank(path2)) {
+            return path1;
+        }
+        var result = path1;
+        if (!result.endsWith("/")) {
+            result += "/";
+        }
+        return result + path2;
+    }
+
+    /**
+     * Builds full repository path.
+     *
+     * @param repoId repository id
+     * @param path   path
+     * @return repository path
+     */
+    public static String buildRepositoryPath(String repoId, String path) {
+        if (StringUtils.isBlank(path)) {
+            return repoId;
+        }
+
+        var parts = path.split("/");
+        var normalizedPath = new StringBuilder(repoId);
+
+        int i = 0;
+        for (var part : parts) {
+            if (!StringUtils.isBlank(part)) {
```

Comment:
There shouldn't be expected empty parts of the path. If the part is empty - it is an error.